GUI/Dialogs/CloneFakeGameDialog.cs

```diff
@@ -144,13 +145,17 @@ private async void buttonOK_Click(object sender, EventArgs e)
 
                 try
                 {
+                    IGame guessedGame = manager.DetermineGame(new DirectoryInfo(existingPath), user);
                     await Task.Run(() =>
                     {
-                        GameInstance sourceInstance = manager.Instances.Values
-                            .FirstOrDefault(i => i.GameDir() == textBoxClonePath.Text);
+                        if (guessedGame == null)
+                        {
+                            throw new NotKSPDirKraken(existingPath);
+                        }
```

Comment:
This can be null not just because there's a problem with the path, but also if the user clicks Cancel in the game selection popup. In that case, the instance is valid but we say it's not.

Reply:
Reworked this to throw `NotKSPDirKraken` from inside `DetermineGame` when we know it's true, then treat null only as cancellation and skip the error message.

Reply:
Yes, I kept it like this because the user will know that they cancelled, so they will understand why they got the message, even if it isn't 100% accurate.

I explicitly avoided throwing an exception inside `DetermineGame`, because we already return `null` in the bad case. Mixing the two isn't good code style IMHO, you'd expect _either_ null for errors _or_ exceptions. Being aware that we commonly overlook the possibility of exceptions being thrown and don't catch them, leading to uncaught ones crashing the whole client, I did decide for `null` here.
If you want to switch to exceptions, please also throw one for the "User cancelled" case, and never return `null`.

Reply:
The user cancelling isn't a bad case or an error, though, it's a normal flow that is available for the user to choose. Whereas the folder not having a game in it is an error.